### ISSUE TITLE
Contributions: Image similarity search engine using triplet loss 

### ISSUE THREAD

**Issue by yasserius** [keras-team/keras-io#271]
Referring to this [call for contribution](https://github.com/keras-team/keras-io/blob/master/call_for_contributions.md#image-similarity-search-engine-using-triplet-loss).

There is already an [example on MNIST](https://www.tensorflow.org/addons/tutorials/losses_triplet), which uses triplet loss.

Now, [Totally Looks Like Dataset](https://sites.google.com/view/totally-looks-like-dataset) provides pairs of images ([left and right matching pairs](https://drive.google.com/drive/folders/1qQJHA5m-vLMAkBfWEWgGW9n61gC_orHl)) rather than class indices like in MNIST. So, the pairs must be preprocessed to be fed into the same triplet loss used in the tutorial above.

If I turn **every pair into its own class index** (so if there are 1000 pairs, then the class labels will be from 0 to 999, with 2 images in each class), should the loss function work?

Or is there a **better way to preprocess the matching pairs**? Please let me know.

Also, is hard or semi hard negative triplet mining relevant in this case, given the distinct left and right images? (compared to similar MNIST images)
The embeddings should be far apart, and we should only have to train the network to bring each pair closer together?

Thanks in advance!

**Comment by sachinprasadhs** [comment]
Hi, 

Thanks for reporting the issue. 

The contribution list is updated. 

For detailed clarification on Triplet loss using Siamese network, you can refer the example mentioned below.

https://keras.io/examples/vision/siamese_network/

Let us know if you still need any further information. Thanks!
